framework/src/wifi_manager/wifi_manager_state.c

```diff
@@ -299,28 +299,30 @@ wifi_manager_result_e _wifimgr_stop_softap(void)
 	return WIFI_MANAGER_SUCCESS;
 }
 
-wifi_manager_result_e _wifimgr_scan(wifi_manager_ap_config_s *config)
+wifi_manager_result_e _wifimgr_scan(wifi_manager_scan_config_s *config)
 {
-	wifi_utils_ap_config_s uconf;
+	trwifi_scan_config_s uconf = {0, {0,}, 0};
 	if (!config) {
 		WIFIMGR_CHECK_UTILRESULT(wifi_utils_scan_ap(NULL), TAG,
-								 "request scan to wifi utils is fail", WIFI_MANAGER_FAIL);
+								 "request full scan is fail", WIFI_MANAGER_FAIL);
 		return WIFI_MANAGER_SUCCESS;
 	}
-
-	strncpy(uconf.ssid, config->ssid, WIFIMGR_SSID_LEN);
-	uconf.ssid[WIFIMGR_SSID_LEN] = '\0';
-	uconf.ssid_length = config->ssid_length;
-	if (config->passphrase_length > 0) {
-		strncpy(uconf.passphrase, config->passphrase, WIFIMGR_PASSPHRASE_LEN);
-		uconf.passphrase[WIFIMGR_PASSPHRASE_LEN] = '\0';
-		uconf.passphrase_length = config->passphrase_length;
+	if (config->ssid_length > 0 && config->channel != 0) {
+		/*  both channel and ssid are set, it's invalid usage */
+		return WIFI_MANAGER_INVALID_ARGS;
+	}
+	if (config->ssid_length > WIFIMGR_SSID_LEN) {
+		return WIFI_MANAGER_INVALID_ARGS;
+	}
+	if (config->ssid_length > 0) {
+		strncpy(uconf.ssid, config->ssid, config->ssid_length + 1);
+		uconf.ssid_length = config->ssid_length;
+	} else if (config->channel != 0) {
```

Comment:
Here, we need to check whether the channel number is valid. It might be negative or big numbers.